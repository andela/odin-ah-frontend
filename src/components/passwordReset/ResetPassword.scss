
.reset-field {
  margin: 10% auto;
  border: 1px solid rgb(3, 96, 172);
  max-height: calc(100vh - 40px);
  max-width: 500px;
  padding: 50px;
}

#resetPasswordBtn{
  margin-top: 20px
}
.reset-field  h1{
  font-weight: bold;
  margin-bottom: 10px;
  font-size: 18px;
  text-align: center
}


#reset-info{
  margin-bottom: 20px;
  text-align: center

}


.alert {
  background-color: rgb(43, 35, 20);
  color:goldenrod;
}
