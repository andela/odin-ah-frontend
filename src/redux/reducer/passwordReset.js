import {
  SAVE_INPUT,
  PASSWORD_RESET_REQUEST,
  PASSWORD_RESET_REQUEST_SUCCESS,
  PASSWORD_RESET_REQUEST_FAILURE,
  COMPLETE_RESET_REQUEST,
  COMPLETE_RESET_REQUEST_SUCCESS,
  COMPLETE_RESET_REQUEST_FAILURE,
  INPUT_ERROR
} from '../constants/resetPassword';

const initialState = {
  email: '',
  password: '',
  confirmPassword: '',
  token: ''
};

const passwordResetReducer = (state = initialState, action) => {
  let newState = { ...state };

  switch (action.type) {
    // eslint-disable-no-case-declarations
    case SAVE_INPUT: {
      const { field, value } = action.payload;
      newState[field] = value;
      return newState;
    }
    case PASSWORD_RESET_REQUEST:
      newState = { ...state };
      newState.resetLoading = true;
      newState.email = '';
      return newState;
    case PASSWORD_RESET_REQUEST_SUCCESS:
      newState = { ...state };
      newState.resetLoading = false;
      newState.message = action.payload.message;
      return newState;
    case PASSWORD_RESET_REQUEST_FAILURE:
      newState = { ...state };
      newState.resetLoading = false;
      newState.message = action.payload.message;
      return newState;
    case COMPLETE_RESET_REQUEST:
      newState = { ...state };
      newState.confirming = true;
      newState.password = '';
      newState.confirmPassword = '';
      return newState;
    case COMPLETE_RESET_REQUEST_SUCCESS:
      newState = { ...state };
      newState.confirming = false;
      newState.message = action.payload.message;
      return newState;
    case COMPLETE_RESET_REQUEST_FAILURE:
      newState = { ...state };
      newState.message = action.payload.message;
      newState.confirming = false;
      return newState;
    case INPUT_ERROR:
      return { ...state, ...action.payload };

    default:
      return state;
  }
};

export default passwordResetReducer;
