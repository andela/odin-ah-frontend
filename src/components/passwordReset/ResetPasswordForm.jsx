import React, { Component } from 'react';
import './ResetPassword.scss';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';

import { saveInput, inputError, completeResetRequest } from '../../redux/actions/resetPassword';
import Alert from '../notification/alert';

export class ResetPasswordForm extends Component {
  render() {
    const {
      password, confirmPassword, confirming, message, errors
    } = this.props.passwordResetData;

    if (message === 'Success Message') return <Redirect to={'/?login'} />;
    const { saveInputHandler, resetRequestHandler } = this.props;
    const { token } = this.props.match.params;
    let passwordError;
    let confirmPasswordError;
    if (errors) {
      passwordError = errors.password;
      confirmPasswordError = errors.confirmPassword;
    }
    return (
      <div>
        <div className="reset-field">
          <Alert />
          <h1>Reset Password Form</h1>
          <form
            onSubmit={(e) => {
              e.preventDefault();
              resetRequestHandler(password, confirmPassword, token);
            }}
          >
            <div className="field">
              <label className="label">Password</label>
              <div className="control has-icons-left has-icons-right">
                <input
                  className={`input is-medium ${passwordError ? 'is-danger' : ''}`}
                  type="password"
                  placeholder="Enter password"
                  value={password}
                  onChange={e => saveInputHandler('password', e.target.value)}
                />
                <span className="icon is-small is-left">
                  <i className="fas fa-key" />
                </span>
                {passwordError && (
                  <span className="icon is-small is-right has-text-danger">
                    <i className="fas fa-exclamation-triangle" />
                  </span>
                )}
              </div>
              {passwordError && <p className="help is-danger">{passwordError[0]}</p>}
            </div>
            <div className="field">
              <label className="label">Confirm Password</label>
              <div className="control has-icons-left has-icons-right">
                <input
                  className={`input is-medium ${confirmPasswordError ? 'is-danger' : ''}`}
                  type="password"
                  placeholder="Enter password"
                  value={confirmPassword}
                  onChange={e => saveInputHandler('confirmPassword', e.target.value)}
                />
                <span className="icon is-small is-left">
                  <i className="fas fa-key" />
                </span>
                {confirmPasswordError && (
                  <span className="icon is-small is-right has-text-danger">
                    <i className="fas fa-exclamation-triangle" />
                  </span>
                )}
              </div>
              {confirmPasswordError && <p className="help is-danger">{confirmPasswordError[0]}</p>}
            </div>

            <button id="resetPasswordBtn"
              className={`button is-primary is-medium is-fullwidth ${
                confirming ? 'is-loading' : ''
              }`}
              type="submit"
              value="Reset password"
            >
              Reset Password
            </button>
          </form>
        </div>
      </div>
    );
  }
}

ResetPasswordForm.propTypes = {
  saveInputHandler: PropTypes.func,
  resetRequestHandlereset: PropTypes.func
};

const mapStateToProps = state => ({
  passwordResetData: state.passwordResetData || {}
});

const mapDispatchToProps = dispatch => ({
  saveInputHandler: (field, value) => {
    dispatch(inputError(null));
    dispatch(saveInput(field, value));
  },
  resetRequestHandler: (password, confirmPassword, token) => {
    dispatch(completeResetRequest({ password, confirmPassword, token }));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ResetPasswordForm);
