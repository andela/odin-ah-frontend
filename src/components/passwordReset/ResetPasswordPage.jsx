import React, { Component } from 'react';
import '@fortawesome/fontawesome-free/css/all.min.css';
import './ResetPassword.scss';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { saveInput, sendResetRequest, inputError } from '../../redux/actions/resetPassword';
import Alert from '../../components/notification/alert';

export class ResetPasswordPage extends Component {
  render() {
    const { email, resetLoading, errors } = this.props.passwordResetData;
    const { saveInputHandler, resetRequestHandler } = this.props;
    let emailError;
    if (errors) {
      emailError = errors.email;
    }
    return (
      <div className="reset-field">
        <Alert />
        <div id="reset-info">
          Enter your email below and we'll send you a link to reset your password.
        </div>
        <form
          id="sendEmailForm"
          onSubmit={(e) => {
            e.preventDefault();
            resetRequestHandler(email);
          }}
        >
          <div className="field">
            <div className="control has-icons-left has-icons-right">
              <input
                // className="input is-medium"
                className={`input is-medium ${emailError ? 'is-danger' : ''}`}
                type="email"
                placeholder="Enter email"
                value={email}
                onChange={e => saveInputHandler('email', e.target.value)}
              />
              <span className="icon is-medium is-left ">
                <i className="fas fa-envelope" />
              </span>
              {emailError && (
                <span className="icon is-small is-right has-text-danger">
                  <i className="fas fa-exclamation-triangle" />
                </span>
              )}
            </div>
            {emailError && <p className="help is-danger">{emailError[0]}</p>}
          </div>

          <button
            id="resetPasswordBtn"
            className={`button is-primary is-medium is-fullwidth ${
              resetLoading ? 'is-loading' : ''
            }`}
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    );
  }
}

ResetPasswordPage.propTypes = {
  saveInputHandler: PropTypes.func,
  resetRequestHandlereset: PropTypes.func
};

const mapStateToProps = state => ({
  passwordResetData: state.passwordResetData || {}
});

const mapDispatchToProps = dispatch => ({
  saveInputHandler: (field, value) => {
    dispatch(inputError(null));
    dispatch(saveInput(field, value));
  },
  resetRequestHandler: (email) => {
    dispatch(sendResetRequest({ email }));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ResetPasswordPage);
