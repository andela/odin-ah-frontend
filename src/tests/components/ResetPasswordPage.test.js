import React from 'react';
import { shallow } from 'enzyme';
import { ResetPasswordPage } from '../../components/passwordReset/ResetPasswordPage';

describe('ResetPassword Component', () => {
  it('ResetPasswordPage', () => {
    const passwordResetData = {
      email: '',
      resetRequestStatus: ''
    };
    const wrapper = shallow(
      <ResetPasswordPage
        passwordResetData={passwordResetData}
        resetRequestHandler={() => jest.fn()}
        saveInputHandler={() => jest.fn()}
      />
    );

    wrapper.find('.input')
      .simulate('change', {
        target: { value: 'hello' }
      });
    const form = wrapper.find('form');
    form.simulate('submit', {
      preventDefault: jest.fn(),

    });

    expect(wrapper)
      .toMatchSnapshot();
  });
  it('ResetPasswordPage', () => {
    const passwordResetData = {
      errors: {
        email: '',
      }
    };
    const wrapper = shallow(
      <ResetPasswordPage
        passwordResetData={passwordResetData}
        resetRequestHandler={() => jest.fn()}
        saveInputHandler={() => jest.fn()}
      />
    );

    expect(wrapper)
      .toMatchSnapshot();
  });
});
