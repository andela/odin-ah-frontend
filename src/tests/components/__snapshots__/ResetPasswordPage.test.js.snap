// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword Component ResetPasswordPage 1`] = `
<div>
  <div
    className="reset-field"
  >
    <Connect(Alert) />
    <div
      id="reset-info"
    >
      Enter your email below and we will send you a link to reset your password.
    </div>
    <form
      id="sendEmailForm"
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <div
          className="control has-icons-left has-icons-right"
        >
          <input
            className="input is-medium "
            onChange={[Function]}
            placeholder="Enter email"
            type="email"
            value=""
          />
          <span
            className="icon is-medium is-left "
          >
            <i
              className="fas fa-envelope"
            />
          </span>
        </div>
      </div>
      <button
        className="button is-primary is-medium is-fullwidth "
        id="resetPasswordBtn"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;

exports[`ResetPassword Component ResetPasswordPage 2`] = `
<div>
  <div
    className="reset-field"
  >
    <Connect(Alert) />
    <div
      id="reset-info"
    >
      Enter your email below and we will send you a link to reset your password.
    </div>
    <form
      id="sendEmailForm"
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <div
          className="control has-icons-left has-icons-right"
        >
          <input
            className="input is-medium "
            onChange={[Function]}
            placeholder="Enter email"
            type="email"
          />
          <span
            className="icon is-medium is-left "
          >
            <i
              className="fas fa-envelope"
            />
          </span>
        </div>
      </div>
      <button
        className="button is-primary is-medium is-fullwidth "
        id="resetPasswordBtn"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
